apps/managedidentity/managedidentity.go

```diff
@@ -434,6 +464,11 @@ func authResultFromToken(authParams authority.AuthParams, token accesstokens.Tok
 	if err != nil {
 		return base.AuthResult{}, err
 	}
+	// if refreshOn is not set, set it to half of the time until expiry if expiry is more than 2 hours away
+	timeUntilExpiry := time.Until(token.ExpiresOn)
+	if token.RefreshOn.T.IsZero() && timeUntilExpiry > (time.Hour*2) {
+		token.RefreshOn.T = time.Now().Add(timeUntilExpiry / 2)
+	}
```

Comment:
please forgive me nitpicking on something with negligible practical impact but this unnecessarily computes and throws away `timeUntilExpiry` when `!token.RefreshOn.T.IsZero()`
```suggestion
	if token.RefreshOn.T.IsZero() {
		if lifetime := time.Until(token.ExpiresOn); lifetime > 2*time.Hour {
			token.RefreshOn.T = time.Now().Add(lifetime / 2)
		}
	}
```

Also, we should test this behavior

Reply:
updated